tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -263,32 +264,20 @@ def _handle_input(self, testing, continue_msg):
                 print('Cancelled.')
                 exit(0)
 
-    def handle(self, *args, **options):
-        """Delete and reparse datafiles matching a query."""
-        fiscal_year = options.get('fiscal_year', None)
-        fiscal_quarter = options.get('fiscal_quarter', None)
-        reparse_all = options.get('all', False)
-        new_indices = reparse_all is True
-
-        # Option that can only be specified by calling `handle` directly and passing it.
-        testing = options.get('testing', False)
-        ##
-
-        args_passed = fiscal_year is not None or fiscal_quarter is not None or reparse_all
-
-        if not args_passed:
-            logger.warn("No arguments supplied.")
-            self.print_help("manage.py", "clean_and_parse")
-            return
-
+    def get_files_to_reparse(case, fiscal_year, fiscal_quarter, selected_files, reparse_all):
+        """Get the files to reparse."""
         backup_file_name = "/tmp/reparsing_backup"
         files = DataFile.objects.all()
         continue_msg = "You have selected to reparse datafiles for FY {fy} and {q}. The reparsed files "
+        if selected_files:
+            files = files.filter(id__in=selected_files)
+            backup_file_name += "_selected_files"
+            continue_msg = continue_msg.format(fy=f"selected files: {str(selected_files)}", q="Q1-4")
         if reparse_all:
             backup_file_name += "_FY_All_Q1-4"
             continue_msg = continue_msg.format(fy="All", q="Q1-4")
```

Comment:
@raftmsohani i selected the most recent version of all files submitted for FY21, tribes only (see below) and observed the continue_msg on line 278 and not the one on line 275. What is the difference between the conditions on 272 and 276? 

![test1](https://github.com/user-attachments/assets/b37fdd81-f941-4f6d-b852-9f9271aad733)
![log_test1](https://github.com/user-attachments/assets/8af78edc-b6cd-450d-9c1d-cf40bc04b713)

Reply:
test # 2 (one file only) results in the same log message

![test2](https://github.com/user-attachments/assets/29d75e77-a788-40ff-87a6-6d45641d0c01)
![log_test2](https://github.com/user-attachments/assets/ee0c800c-3fbd-48e7-9220-2fa1e769559a)

Reply:
i think you're actually getting the log from line 350

```python
log(f"Starting clean and reparse command for FY {fiscal_year if fiscal_year else all_fy} and "
            f"{fiscal_quarter if fiscal_quarter else all_q}",
            logger_context=log_context,
            level='info')
```

which shows `ALL` if a fiscal year/quarter is not specified. similar to your other comment, we don't actually pass those values into the reparse command when running it via the action, only if they are provided in the management commend args.

#3205 talks about refactoring the action such that it is no longer coupled with the management command.